pkg/didcomm/protocol/didexchange/event.go

```diff
@@ -14,3 +14,30 @@ type Event interface {
 	// invitation ID
 	InvitationID() string
 }
+
+// didExchangeEvent implements didexchange.Event interface.
+type didExchangeEvent struct {
+	connectionID string
+	invitationID string
+}
+
+// ConnectionID returns DIDExchange connectionID.
+func (ex *didExchangeEvent) ConnectionID() string {
+	return ex.connectionID
+}
+
+// InvitationID returns DIDExchange invitationID.
+func (ex *didExchangeEvent) InvitationID() string {
+	return ex.invitationID
+}
+
+// didExchangeEvent for sending events with processing error.
+type didExchangeEventError struct {
```

Comment:
Do not understand why we need `didExchangeEventError` structure, we can keep an error in `didExchangeEvent`

Reply:
didExchangeEventError - implements error interface. This is done as client can differentiate between type ie, error or not. The base struct provides protocol specific info like connectionID and invitationID.

Reply:
yeah I saw, but my point is that error can also be determined by calling `.Err() error` function, it also can be like protocol-specific info

Reply:
Yeah, I think it's design decision based on how we want to do it - validating against type or .Err()!= nil.

Reply:
Validating against type is better.

Create a specific data structure for a particular codepath. (i'm not a fan of optional fields that are only filled in a certain codepath.)